magento/magento2
# Update box model to border box

### Description (*)

Why we need update
box-sizing supported by all major browsers. Later allow make and craft markup grid easier

We switch the global box-sizing value to border-box. This ensures padding does not affect the final computed width of an element

#1 Update box-model to border-box 
Magento have define .lib-set-default-border-box but it's never used in code base
When use box-sizing border box we make frontend dev life easier. Only define in one file and styles apply to all other place for all element.
we can use content-box or padding-box (where supported) as well, without worrying about a universal selector overriding css

box-sizing border box defined in file app/design/frontend/Magento/blank/web/css/source/_reset.less

<img width="309" alt="Screen Shot 2020-12-03 at 10 38 15" src="https://user-images.githubusercontent.com/1908873/100960860-b9c22c00-3553-11eb-9d9d-352812cca2ae.png">

https://thisthat.dev/border-box-vs-content-box
Now a lot of web use border-box model as default

When border-box is used border,  padding-left and padding-right are not included in the calculation box width

all elements can re-define box model (content-box) without any issue if need. Default all elements in html will have border box model

This PR only remove define border-box in theme core/libs magento in luma theme
Keep old declare mixins related boxder-box for maintain compatible in 3rd-party themes
Keep backward compatible in blank theme (avoid too much changes in blank)

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. Check all pages blank/luma theme make sure all page work without break, glitch, overlap box


### Questions or comments

Old article related
https://css-tricks.com/inheriting-box-sizing-probably-slightly-better-best-practice
Need workaround check visual both luma/blank theme when apply border-box model
Should test all screen make sure everything work like before no big visual change break any elements

CC: @krzksz @Leland @ptylek  good to hear feedback from you guys
@DanielRuf ,@Parakoopa, @kirmorozov ,@leoquijano, @jonathanKingston can you guys give me some opinion

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#31137: Update box model to border box
